sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/ui/ThriftServerPage.scala

```diff
@@ -135,11 +141,15 @@ private[ui] class ThriftServerPage(parent: ThriftServerTab) extends WebUIPage(""
   }
 
   /** Generate stats of batch sessions of the thrift server program */
-  private def generateSessionStatsTable(): Seq[Node] = {
-    val numBatches = listener.sessionList.size
+  private def generateSessionStatsTable(online: Boolean): Seq[Node] = {
+    val numBatches = if(online) listener.onlineSessionNum else (listener.sessionList.size-listener.onlineSessionNum)
     val table = if (numBatches > 0) {
-      val dataRows =
-        listener.sessionList.values
+      val dataRowsTmp = listener.sessionList.values.toSeq
+      val dataRows = if(online) {
+        dataRowsTmp.filter(_.finishTimestamp == 0).sortBy(_.startTimestamp).reverse
```

Comment:
You don't need to sort by the `startTimestamp` because the list is a `LinkedHashMap`